internal/processmon/processmon.go

```diff
@@ -0,0 +1,379 @@
+	req.Payload = s.(*processInfo).process.Pid
+
+	c := make(chan error, 1)
+	go func() {
+		c <- s.(*processInfo).RPCHdl.RemoteCall(contextID, "Server.EnforcerExit", req, resp)
+	}()
+	select {
+	case kerr := <-c:
+		if kerr != nil {
+			zap.L().Debug("Failed to stop gracefully",
+				zap.String("Remote error", kerr.Error()))
+		}
+
+		if perr := s.(*processInfo).process.Kill(); perr != nil {
+			zap.L().Debug("Process is already dead",
+				zap.String("Kill error", perr.Error()))
+		}
+
+	case <-time.After(5 * time.Second):
+		if perr := s.(*processInfo).process.Kill(); perr != nil {
+			zap.L().Info("Time out while killing process ",
+				zap.Error(perr))
+		}
+	}
+
+	s.(*processInfo).RPCHdl.DestroyRPCClient(contextID)
+	if err := os.Remove(filepath.Join(p.netNSPath, contextID)); err != nil {
+		zap.L().Warn("Failed to remote process from netns path", zap.Error(err))
+	}
+
+	if err := p.activeProcesses.Remove(contextID); err != nil {
+		zap.L().Warn("Failed to remote process from cache", zap.Error(err))
+	}
+}
+
+// pollStdOutAndErr polls std out and err
+func (p *processMon) pollStdOutAndErr(
+	cmd *exec.Cmd,
+	exited chan int,
+	contextID string,
+) (initializedCount int, err error) {
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return initializedCount, err
+	}
+
+	initializedCount++
+
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return initializedCount, err
+	}
+
+	initializedCount++
+
+	// Stdout/err processing
+	go processIOReader(stdout, contextID, exited)
+	go processIOReader(stderr, contextID, exited)
+
+	return initializedCount, nil
+}
+
+// getLaunchProcessCmd returns the command used to launch the enforcerd
+func (p *processMon) getLaunchProcessCmd(arg string, contextID string) (*exec.Cmd, error) {
+
+	cmdName, err := osext.Executable()
+	if err != nil {
+		return nil, err
+	}
+
+	cmdArgs := []string{arg}
+	cmdArgs = append(cmdArgs, p.launcProcessArgs...)
+	if !p.logToConsole {
+		cmdArgs = append(cmdArgs, contextID)
+	}
+	zap.L().Debug("Enforcer executed",
+		zap.String("command", cmdName),
+		zap.Strings("args", cmdArgs),
+	)
+
+	return exec.Command(cmdName, cmdArgs...), nil
+}
+
+// getLaunchProcessEnvVars returns a slice of env variable strings where each string is in the form of key=value
+func (p *processMon) getLaunchProcessEnvVars(
+	procMountPoint string,
+	contextID string,
+	randomkeystring string,
+	statsServerSecret string,
+	refPid int,
+	refNSPath string,
+) []string {
+
+	mountPoint := constants.AporetoEnvMountPoint + "=" + procMountPoint
+	namedPipe := constants.AporetoEnvContextSocket + "=" + contextID2SocketPath(contextID)
+	statsChannel := constants.AporetoEnvStatsChannel + "=" + rpcwrapper.StatsChannel
+	rpcClientSecret := constants.AporetoEnvRPCClientSecret + "=" + randomkeystring
+	envStatsSecret := constants.AporetoEnvStatsSecret + "=" + statsServerSecret
+	containerPID := constants.AporetoEnvContainerPID + "=" + strconv.Itoa(refPid)
+
+	newEnvVars := []string{
```

Comment:
if you only used them here, don't assign variable, just do your `+` in here

Reply:
done